File: BEmu/IntradayTickRequest/IntradayTickRequestElementInt.cs
//------------------------------------------------------------------------------
// <copyright project="BEmu_csh" file="IntradayTickRequest/IntradayTickRequestElementInt.cs" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

namespace Bloomberglp.Blpapi.IntradayTickRequest
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;

    internal class IntradayTickRequestElementInt : IntradayTickRequestElementString
    {
        private readonly int _value;

        internal IntradayTickRequestElementInt(string elementName, int value) : base(elementName, value.ToString())
        {
            this._value = value;
        }

        //I can't override GetElementAsInt32 here because the Bloomberg Request object stores ints as strings, not ints.  You can't convert the string to an int
        internal int GetInt { get { return this._value; } }
        public override Schema.Datatype Datatype { get { return Schema.Datatype.INT32; } }
    }
}

File: BEmu_cpp/src/IntradayTickRequest/IntradayTickRequestElementInt.cpp
//------------------------------------------------------------------------------------------------
// <copyright project="BEmu_cpp" file="src/IntradayTickRequest/IntradayTickRequestElementInt.cpp" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------------------------

#include "IntradayTickRequest/IntradayTickRequestElementInt.h"
#include "IntradayTickRequest/IntradayTickRequestElementString.h"
#include "BloombergTypes/Name.h"
#include "BloombergTypes/Element.h"
#include "Types/IndentType.h"

namespace BEmu
{
	namespace IntradayTickRequest
	{
		IntradayTickRequestElementInt::IntradayTickRequestElementInt(const std::string& elementName, int ivalue) :
			IntradayTickRequestElementString(elementName, ivalue)
		{
			this->_ivalue = ivalue;
		}

		IntradayTickRequestElementInt::~IntradayTickRequestElementInt()
		{
		}

		int IntradayTickRequestElementInt::getInt() const
		{
			return this->_ivalue;
		}
	}
}